eng/common/scripts/Generate-DocIndex.ps1

```diff
@@ -0,0 +1,89 @@
+            $serviceName = "Other"
+        }
+        else {
+            if ($packageInfo.Length -gt 1) {
+                LogWarning "There are more than 1 packages fetched out for artifact $artifact. Please check csv of Azure/azure-sdk/_data/release/latest repo if this is intended. "
+            }
+            $serviceName = $packageInfo[0].ServiceName.Trim()
+        }
+        $orderServiceMapping[$artifact] = $serviceName
+    }
+    return $orderServiceMapping                   
+}
+
+function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang) {
+    LogDebug "Start generating the docfx toc and build docfx site..."
+    $DocOutDir = "${RepoRoot}/docfx_project"
+
+    LogDebug "Initializing Default DocFx Site..."
+    & $($DocFx) init -q -o "${DocOutDir}"
+    # The line below is used for testing in local
+    # docfx init -q -o "${DocOutDir}"
+    LogDebug "Copying template and configuration..."
+    New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory" -Force
+    Copy-Item "${DocGenDir}/templates/*" -Destination "${DocOutDir}/templates" -Force -Recurse
+    Copy-Item "${DocGenDir}/docfx.json" -Destination "${DocOutDir}/" -Force
+    $YmlPath = "${DocOutDir}/api"
+    New-Item -Path $YmlPath -Name "toc.yml" -Force
+    $visitedService = @{}
+    # Sort and display toc service name by alphabetical order.
+    foreach ($serviceMapping in $tocContent.getEnumerator() | Sort Value) {
+        $artifact = $serviceMapping.Key
+        $serviceName = $serviceMapping.Value
+        $fileName = ($serviceName -replace '\s', '').ToLower().Trim()
+        if ($visitedService.ContainsKey($serviceName)) {
+            Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact"
+        }
+        else {
+            Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: ${serviceName}`r`n  href: ${fileName}.md"
+            New-Item -Path $YmlPath -Name "${fileName}.md" -Force
+            Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact"
+            $visitedService[$serviceName] = $true
+        }
+    }
+
+    # Generate toc homepage.
+    LogDebug "Creating Site Title and Navigation..."
+    New-Item -Path "${DocOutDir}" -Name "toc.yml" -Force
+    Add-Content -Path "${DocOutDir}/toc.yml" -Value "- name: Azure SDK for $lang APIs`r`n  href: api/`r`n  homepage: api/index.md"
+
+    LogDebug "Copying root markdowns"
+    Copy-Item "$($RepoRoot)/README.md" -Destination "${DocOutDir}/api/index.md" -Force
+    Copy-Item "$($RepoRoot)/CONTRIBUTING.md" -Destination "${DocOutDir}/api/CONTRIBUTING.md" -Force
+
+    LogDebug "Building site..."
+    & $($DocFx) build "${DocOutDir}/docfx.json"
+    # The line below is used for testing in local
+    # docfx build "${DocOutDir}/docfx.json"
+    Copy-Item "${DocGenDir}/assets/logo.svg" -Destination "${DocOutDir}/_site/" -Force    
+}
+
+&$GetGenerateGithubIoDocIndexFn
```

Comment:
Lets add some checks similar to https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/copy-docs-to-blobstorage.ps1#L234 to ensure this function exists.